sound/soc/sof/intel/hda-trace.c

```diff
@@ -26,7 +26,12 @@ static int hda_dsp_trace_prepare(struct snd_sof_dev *sdev, struct snd_dma_buffer
 	struct hdac_stream *hstream = &hext_stream->hstream;
 	int ret;
 
-	hstream->period_bytes = 0;/* initialize period_bytes */
+	/* initialize period_bytes */
+	/* use a low period size to work with firmware versions
+	 * that do not send IPC messages when traces are transferred
+	 * via DMA and we need to rely on HDA DMA interrupts
+	 */
+	hstream->period_bytes = 256;
```

Comment:
should this be set by looking at dmab->bytes? e.g. dmab->bytes/4 or something.

I am not sure why this initialization was needed. in the original commit b846ee5671e8e, the comment was 

"
Without this initialization, the first_boot works fine, but Code Loader
    or DMA Trace may failed after suspend/resume, as the stream might be used
    by other streams with different period_bytes setting before suspend.
"

Reply:
@plbossart With these FW builds, we don't get any IPC notifications about data transferred, so we need to set up period interrupts for trace as well. By setting this to zero, we only get one interrupt when the whole 64KiB is transferred. The 256 setting is a bit slow as well, so this will need more work to tune.